### PR TITLE
Support HttpRequestMessage.Version in WinHttpHandler

### LINKED ISSUE #4754: HttpClient doesn't respect HttpRequestMessage.Version
e.g. if it's set to 1.0, the request still gets sent with a version 1.1

### PR DESCRIPTION
**Pull request by davidsh**
WinHttpHandler was currently ignoring the `HttpRequestMessage.Version`
property and was always sending out requests using 'HTTP/1.1' which is the
default for winhttp.

This fix now will look at the request message `Version` property and
send out the request using either 1.0 or 1.1.  If neither of these
versions are specified, then the request is sent out using the default
winhttp version (which is currently 1.1 as of Windows 10).

Note: The .NET Framework (Desktop) will throw an exception if neither
1.0 nor 1.1 version is specified.  Currently CoreFx including NETNative,
WinHttpHandler and CurlHandler will simply use a default instead of
throwing. While this is different behavior than Desktop, it is more
desirable in the long run.

There is a separate issue #4870 to add HTTP/2.0 support to WinHttpHandler.

Fixes #4754

### CODE REVIEW

**Comment by davidsh** [comment]
@CIPop @stephentoub @SidharthNabar PTAL

File: src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs [hunk 1]
```diff
@@ -791,12 +791,24 @@ private void EnsureSessionHandleExists(WinHttpRequestState state)
                     secureConnection = false;
                 }
 
+                // Try to use the requested version if a known/supported version was explicitly requested.
```
Comment (CIPop) [inline src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs]: Nit: Why not extract a method here?

**Comment by stephentoub** [comment]
LGTM

**Comment by CIPop** [comment]
LGTM

File: src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs (hunk 1 above)
Comment (davidsh) [inline src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs]: It's seem too small at this point to add another method for just that. But I will re-consider later if this code block grows further.

**Comment by davidsh** [comment]
Infrastructure issues with OSX CI:
>Connectivity Issue
Possible issue connecting to GitHub or cloning the repository. This could be caused by a lack of network connectivity or a lack of disk space. Please email netciadmin

**Comment by stephentoub** [comment]
We've been having bad infrastructure issues with regards to OS X.  Feel free to merge.

**Pull request merged**